src/Avalonia.Base/Data/Core/IndexerNodeBase.cs

```diff
@@ -84,6 +86,6 @@ private bool ShouldUpdate(object sender, NotifyCollectionChangedEventArgs e)
             return true; // Implementation defined meaning for the index, so just try to update anyway
         }
 
-        protected abstract bool ShouldUpdate(object sender, PropertyChangedEventArgs e);
+        protected abstract bool ShouldUpdate(object? sender, PropertyChangedEventArgs e);
```

Comment:
I think it will be better if "sender" notifications will be filtered-out inside of StartListeningCore, so overrides won't need to care about null. Anyway it's hard to imagine null sender for INPC/INCC, and it's most likely only because of universal EventHandler definition.

Reply:
Possibly yeah but that would be a breaking change.